### PR TITLE
Added select statements and renamed column name in control

### PR DESCRIPTION
**Pull request by warmlund**
Added workbench file consistitng of a erd diagram based on our tables.

### CODE REVIEW

File: select_statements.sql [hunk 1]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
 
 -- Underhållspersonal
 -- Sök fram alla bilar i behov av kontroll
+SELECT vehicle.registrationNumber FROM green_rental.vehicle
+LEFT JOIN booking_details ON vehicle.registrationNumber = booking_details.registrationNumber
```
Comment (LarsCJensen) [inline select_statements.sql]: Should be INNER JOIN

File: select_statements.sql [hunk 2]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
...
+LEFT JOIN booking ON booking_details.bookingNumber = booking.bookingNumber
+LEFT JOIN control ON vehicle.registrationNumber=control.registrationNumber
+WHERE control.controlLarge is FALSE and control.datum < "2024-04-12" and booking.endDate = "2024-04-12";
```
Comment (LarsCJensen) [inline select_statements.sql]: Predicate on controlLarge is not needed.

File: select_statements.sql [hunk 3]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
...
+
 -- Sök fram alla bilar i behov av kontroll inom 3/6/12 månader
+SELECT vehicle.registrationNumber
+FROM green_rental.vehicle
+LEFT JOIN control ON control.registrationNumber=vehicle.registrationNumber
+WHERE control.controlLarge is TRUE and control.datum>=DATE_SUB(CURDATE(), INTERVAL 3 MONTH);
```
Comment (LarsCJensen) [inline select_statements.sql]: Should also have control.datum is NULL.
Should probably be less than date + 3 month.

Could perhaps use https://www.w3schools.com/mysql/func_mysql_date_add.asp

File: select_statements.sql [hunk 4]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
...
+
 -- Sök fram alla bilar som har en skada
+SELECT vehicle.RegistrationNumber, damage.damageID, damage.descriptionDamage
+FROM green_rental.vehicle
+LEFT JOIN control ON control.registrationNumber=vehicle.registrationNumber
+LEFT JOIN damage ON damage.controlID=control.controlID
```
Comment (LarsCJensen) [inline select_statements.sql]: Should be INNER JOIN.

File: select_statements.sql [hunk 5]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
...
+WHERE damageID is not NULL AND damage.repairedDate is NULL;
```
Comment (LarsCJensen) [inline select_statements.sql]: If you have INNER JOIN this predicate (damageId) is not needed.

File: select_statements.sql [hunk 6]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
...
+
 -- Lägg till en skada
+INSERT INTO green_rental.damage (controlID,fixedDamage,repairedDate,descriptionDamage)
+SELECT c.controlID,
+FALSE as fixedDamage,
+NULL as repairedDate,
+"Stenskott" as descriptionDamage
+FROM green_rental.control as c
+WHERE c.registrationNumber="ABC123";
```
Comment (LarsCJensen) [inline select_statements.sql]: This SELECT can return multiple hits which would cause it to fail.

File: select_statements.sql [hunk 7]
```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
...
+INSERT INTO green_rental.control (registrationNumber,staffID,controlDate,controlLarge,fuelLevel)
+SELECT v.registrationNumber,
+2 as staffID,
+b.endDate as controlDate,
+FALSE as controlLarge,
+30 as fuelLevel
+FROM green_rental.vehicle as v
+LEFT JOIN booking_details bd ON v.registrationNumber=bd.registrationNumber
+LEFT JOIN booking b ON bd.bookingNumber=b.bookingNumber
+WHERE b.endDate="2024-04-12";
```
Comment (LarsCJensen) [inline select_statements.sql]: This is likely not needed.

**Review by LarsCJensen — CHANGES_REQUESTED** [review]

File: select_statements.sql (hunk 1 above)
Comment (warmlund) [reply select_statements.sql]: Fixed

File: select_statements.sql (hunk 2 above)
Comment (warmlund) [reply select_statements.sql]: Fixed

File: select_statements.sql (hunk 4 above)
Comment (warmlund) [reply select_statements.sql]: Fixed

File: select_statements.sql (hunk 5 above)
Comment (warmlund) [reply select_statements.sql]: Fixed

File: select_statements.sql (hunk 7 above)
Comment (warmlund) [reply select_statements.sql]: Replaced with manually added values instead

File: select_statements.sql [hunk 8]
```diff
@@ -38,44 +38,31 @@ WHERE v.stationName = "Uppsala station" AND vc.name = "Stadsbil" AND (b.endDatum
...
-WHERE control.controlLarge is FALSE and control.datum < "2024-04-12" and booking.endDate = "2024-04-12";
+WHERE control.datum < "2024-04-12" and booking.endDate = "2024-04-12";
 
 -- Sök fram alla bilar i behov av kontroll inom 3/6/12 månader
-SELECT vehicle.registrationNumber
+SELECT vehicle.registrationNumber as Registreringsnummer
 FROM green_rental.vehicle
 LEFT JOIN control ON control.registrationNumber=vehicle.registrationNumber
-WHERE control.controlLarge is TRUE and control.datum>=DATE_SUB(CURDATE(), INTERVAL 3 MONTH);
+WHERE control.controlLarge is TRUE or control.controlDate is NULL and CURDATE()>=DATE_ADD(control.controlDate, INTERVAL 3 MONTH);
```
Comment (LarsCJensen) [inline select_statements.sql]: I think what is even better is to use control.controlDate <= DATE_ADD(CURDATE(), INTERVAL 3 MONTH)

File: select_statements.sql [hunk 9]
```diff
@@ -38,44 +38,31 @@ WHERE v.stationName = "Uppsala station" AND vc.name = "Stadsbil" AND (b.endDatum
...
 FROM green_rental.vehicle
-LEFT JOIN control ON control.registrationNumber=vehicle.registrationNumber
-LEFT JOIN damage ON damage.controlID=control.controlID
-WHERE damageID is not NULL AND damage.repairedDate is NULL;
+INNER JOIN control ON control.registrationNumber=vehicle.registrationNumber
+INNER JOIN damage ON damage.controlID=control.controlID
+WHERE damage.repairedDate is NULL;
 
 -- Lägg till en skada
 INSERT INTO green_rental.damage (controlID,fixedDamage,repairedDate,descriptionDamage)
```
Comment (LarsCJensen) [inline select_statements.sql]: If you want to, you could skip repairedDate. It will get NULL as value.

File: select_statements.sql [hunk 10]
```diff
@@ -38,44 +38,31 @@ WHERE v.stationName = "Uppsala station" AND vc.name = "Stadsbil" AND (b.endDatum
...
-SELECT v.registrationNumber,
-2 as staffID,
-b.endDate as controlDate,
-FALSE as controlLarge,
-30 as fuelLevel
-FROM green_rental.vehicle as v
-LEFT JOIN booking_details bd ON v.registrationNumber=bd.registrationNumber
-LEFT JOIN booking b ON bd.bookingNumber=b.bookingNumber
-WHERE b.endDate="2024-04-12";
+VALUES("ABC123",1,2024-04-21,FALSE,30);
```
Comment (LarsCJensen) [inline select_statements.sql]: I actually didn't think you could use date like that, without quotes. Does it work?

**Review by LarsCJensen — CHANGES_REQUESTED** [review]

File: select_statements.sql (hunk 10 above)
Comment (warmlund) [reply select_statements.sql]: nah I think I just messed up. Fixed now

File: select_statements.sql (hunk 9 above)
Comment (warmlund) [reply select_statements.sql]: Fixed

File: select_statements.sql (hunk 8 above)
Comment (warmlund) [reply select_statements.sql]: Nice, fixed!

**Review by LarsCJensen — APPROVED** [review]

**Pull request merged**